Require C++17

cticenhour opened the pull request:
This PR removes entries for the C++11 standard in compiler flags and replaces it with C++17. We should be covered during compilation by the definition of CXXFLAGS in our most-recent `moose-mpich` package, but this helps ensure that the default standard is C++17 for those who do not use our packages when compiling hit, using `clang_autocomplete`, and in the `MooseSourceParser`. 

Refs #20377 and #20424

cticenhour committed:
Remove entries for C++11 standard and replace with C++17 (#20377)

cticenhour commented:
@loganharbour we can merge this or leave it if you want. I just did a search and found these during our earlier discussion on Slack.

loganharbour reviewed (changes requested):
I'm not sure if we're ready for this. We need a libmesh update to be able to enforce c++17 first, and we also need to change the requirements in documentation.

loganharbour commented:
@cticenhour wanna add a submodule update to https://github.com/libMesh/libmesh/commit/617928abf97fb44d6af0970b91c883e1428e1e8a in libmesh? And then update the docs?

cticenhour commented:
Yeah, I can do that. Might not be today, as I plan to take part of the afternoon as PL, but by Monday for sure.

loganharbour commented:
No worries, I'll get to it this afternoon then 👍🏼

loganharbour committed:
Update libmesh

Summary of changes:

- Add triangle tests
- Optimize RATIONAL_BERNSTEIN evaluations for IGA problems
- Fix unistd.h issues when building with MS VC
- Enable EIM using POD
- outer_product: scalar * vector -> vector
- Detect PETSc HDF5 when --enable-hdf5
- Implement doubly-checked lock for _get_array
- Use range-based for-loops in misc_ex15
- Treat *.C files as C++
- Use HAVE_SYS_TIME_H, HAVE_DIRECT_H, HAVE_PROCESS_H, HAVE_MKSTEMP, HAVE_GETTIMEOFDAY
- TriangulatorInterface:: refactor
- Export explicit template instantiations when building with MS VC
- Differentiate tasks for InFEM (InFEM::reinit)
- Support configure --with-metaphysicl* directory specifications
- Return the 3x3 rotation matrix computed in MeshTools::Modification::rotate()
- ExodusII_IO::write_nodeset_data() should take const refs
- Add some support for Bezier elements in VTK output.
- Add a VTK check before trying to write out a VTK file
- fix error in non-deprecated variant
- Allow use of matrix-free when computing resid and jac together
- Clang fixes
- Mesh refinement support in Poly2TriTriangulator
- Protect access to libMesh::cerr
- More robust Packing of trivially-copyable classes
- Misc configuration fixes
- Set PETSc mat/vec types at runtime
- Fix CI Mac builds
- Fixes for --disable-optional builds
- Fixes for --disable-amr builds
- DistributedMesh fixes for new unit tests, new triangulator
- Require C++17 support
- Some example threading fixes
- Add static TensorValue::rotation_matrix helper
- Make adjoint ex5 benchmark pass
- Lock against concurrent read-write/write-write in Vector::add/insert/set

refs #000

loganharbour committed:
Change c++ requirement in doco c++14 -> c++17

loganharbour reviewed (changes requested):
We still need to address https://github.com/idaholab/moose/blob/next/framework/doc/packages_config.yml; haven't gotten around to it

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 544e2c87ea2f35aab900dde8b8f1e0926d720421
+Subproject commit 617928abf97fb44d6af0970b91c883e1428e1e8a
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/967194/) on 3a4fd8d wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/967194/) on 3a4fd8d wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/967220/) on 3a4fd8d wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/20397/timing)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/967215/) on 3a4fd8d wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.34%</td>
      <td>82.34%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>73848</td>
      <td>73862</td>
      <td>+14</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15840</td>
      <td>15841</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/framework)


## Modules coverage

### Contact

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.03%</td>
      <td>89.04%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>2841</td>
      <td>2842</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>350</td>
      <td>350</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/contact.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/contact)

### Heat conduction

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.52%</td>
      <td>89.50%</td>
      <td>-0.03%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>2991</td>
      <td>2991</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>350</td>
      <td>351</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/heat_conduction.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/heat_conduction)

### Navier stokes

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>83.61%</td>
      <td>83.62%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>8348</td>
      <td>8349</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1636</td>
      <td>1636</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/navier_stokes.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/navier_stokes)

### Peridynamics

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>79.20%</td>
      <td>79.10%</td>
      <td>-0.10%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3039</td>
      <td>3036</td>
      <td>-3</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>798</td>
      <td>802</td>
      <td>+4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/peridynamics.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/peridynamics)

### Phase field

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.21%</td>
      <td>90.18%</td>
      <td>-0.03%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>11973</td>
      <td>11984</td>
      <td>+11</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1300</td>
      <td>1305</td>
      <td>+5</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/phase_field.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/phase_field)

### Porous flow

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>96.23%</td>
      <td>96.23%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>10531</td>
      <td>10544</td>
      <td>+13</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>413</td>
      <td>413</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/porous_flow.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/porous_flow)

### Ray tracing

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>94.96%</td>
      <td>94.96%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>4088</td>
      <td>4091</td>
      <td>+3</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>217</td>
      <td>217</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/ray_tracing.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/ray_tracing)

### Richards

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>97.64%</td>
      <td>97.61%</td>
      <td>-0.03%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3346</td>
      <td>3346</td>
      <td>-</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>81</td>
      <td>82</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/richards.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/richards)

### Tensor mechanics

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>89.44%</td>
      <td>89.44%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>26169</td>
      <td>26179</td>
      <td>+10</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>3090</td>
      <td>3092</td>
      <td>+2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/tensor_mechanics.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/tensor_mechanics)

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.46%</td>
      <td>90.47%</td>
      <td>+0.01%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>14694</td>
      <td>14698</td>
      <td>+4</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1550</td>
      <td>1548</td>
      <td>-2</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/thermal_hydraulics.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/thermal_hydraulics)

### Xfem

<table>
  <thead>
    <tr>
      <th></th>
      <th>bc2965</th>
      <th colspan=4>#20397 3a4fd8</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>81.74%</td>
      <td>81.77%</td>
      <td>+0.03%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>6289</td>
      <td>6299</td>
      <td>+10</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1405</td>
      <td>1404</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage_diff/xfem.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20397/coverage/xfem)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20397/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour commented:
Nice. Looks like this is good to go other than https://github.com/idaholab/moose/pull/20397#pullrequestreview-892950550.

milljm reviewed (dismissed):
👍

roystgnr commented:
What all is left to be done here?  Just bumping up to `minimum_gcc: 7.5.0`, `minimum_clang: 5.0.2`, `minimum_intel: 19.0.1` in packages_config.yml?

loganharbour commented:
@roystgnr nothing is left now.

loganharbour reviewed (dismissed):
Approving changes that are not my own.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/970374/) on abd6953 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/970374/) on abd6953 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr commented:
I'm pretty confident that this PR didn't cause `test:pressure_bc.inclined_edge_2d_pressure FAILED (EXODIFF)` in Valgrind xfem, but all the other tests are `MooseDocs.common.exceptions.MooseDocsException: The following command combination is unknown: 'package minimum_clang'.` and that actually looks like it might be our fault...

cticenhour commented:
Yeah, Logan discovered late last night that removing the whole package extension might be premature considering that it has been useful in propagating our minimum requirements through the documentation. The other results might be the random failures we've been seeing lately.

roystgnr commented:
Yeah, that's our fault.  Assuming we really want to go through with #20424 (and do we? maybe "we" manage our packages with conda, but it'd be really nice if that was a convenience not a requirement; "apt install mooseframework" would be nice to do someday and it's never going to happen if it requires conda), we need to grep through the remaining docs for `minimum_clang` etc. and replace those invocations.

cticenhour commented:
We went back and forth with it yesterday; an ideal world to me would be tying the versions of our conda dependencies to the documentation in a similar way as the package extension did with the old `moose-environment` package. One problem with keeping the current extension around as it is....it affects our manual installation instructions (which may be woefully out of date at this point) and just de-couples things from our well-tested configurations.

loganharbour commented:
I do plan on retaining the package extension now - it’ll just store less information (compilers, mpich, petsc, maybe libmesh). I’ll get around to that today.

lindsayad commented:
Anything I can do to help here? I'd be inclined to limit that xfem test to one timestep as it's doing everything interesting in the first timestep anyways and then after that there's barely anything to solve

loganharbour commented:
I'm finishing up the documentation stuff right now; should be good soon. Not sure what's going on with xfem... doesn't look deterministic

lindsayad commented:
It's probably not, that's why I think we should change the test at some point (doesn't have to be here)

loganharbour committed:
Remove moose_packages in packages MooseDocs extension, bring config up to date

Refs #20424

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/972941/) on adddd4c wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/972941/) on adddd4c wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour committed:
Add newsletter entry for c++17 support and libmesh update

cticenhour commented on modules/doc/content/newsletter/2022_03.md:
```diff
@@ -7,6 +7,22 @@ for a complete description of all MOOSE changes.
 
 ## MOOSE Improvements
 
+### C++17 Support
+
+MOOSE now requires a C++17 capable compiler and supports C++17 features. With this, GCC 7 and Clang 5 are now the minimum supported compilers.
+
 ## libMesh-level Changes
 
+- Work towards source code compatibility with Windows
+- Add an additional `outer_product` definition that supports `scalar * vector -> vector`
+- Add auto-detection for PETSc external packages, notably HDF5
+- Add internal thread safety within `NumericVector` subclasses
+- Add initial support for `Poly2Tri` triangulations and refactor `TriangulatorInterface`
+- Optimize `InfFE::reinit` when not all data has been requested
+- Support the matrix-free case when computing the residual and Jacobian together
+- Add more robust `Packing` for communication of fixed-size classes win variable-size containers
+- Enable setting PETSc mat/vec types at runtime for future GPU use
+- Require C++17 support and begin using it internally
+- Add new static `TensorValue::rotation_matrix` helper functions
+
```
LGTM! 👍🏻

loganharbour reviewed (approved):
Approving changes that are not my own.

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 544e2c87ea2f35aab900dde8b8f1e0926d720421
+Subproject commit a7279e6a5f0ea1f621d894c55a771197241f8708
```
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/973423/) on f6a5796 wanted to post the following:

The following file(s) are changed:

  conda/mpich/conda_build_config.yaml

The conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/973423/) on f6a5796 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops
